eventmesh-storage-plugin/eventmesh-storage-kafka/src/main/java/org/apache/eventmesh/storage/kafka/producer/ProducerImpl.java

```diff
@@ -42,36 +41,20 @@
 
 @Slf4j
 @SuppressWarnings("deprecation")
-public class ProducerImpl {
+public class ProducerImpl extends AbstractProducer {
 
     private final KafkaProducer<String, CloudEvent> producer;
-    private final Properties properties = new Properties();
-    private final AtomicBoolean isStarted = new AtomicBoolean(false);
 
     public ProducerImpl(Properties props) {
+        super(props);
+        properties.clear();
```

Comment:
why use `clear()` method at here

Reply:
Because `super(props)` assigns param `props` to field `properties`. but `ProducerImpl` does not assign value like that. What it do to `properties` is just the next 3 line.
https://github.com/apache/eventmesh/blob/470329d7e3c69fa241762b364f1cd1799a3b5766/eventmesh-storage-plugin/eventmesh-storage-kafka/src/main/java/org/apache/eventmesh/storage/kafka/producer/ProducerImpl.java#L51-L54

Reply:
Excuse me but I still don't understand that why Kafka needs a `clear()` statement here but RocketMQ and Pulsar don't.

Reply:
Then I explain it in detail again.

The following is the original logic of this method under Kafka module. You can see that only three items need to be set in `properties` object. Now after the unified abstraction, `properties` is first assigned a value, which is equivalent to setting more items. If I don't `clear()` first, it firstly disrupts the original logic, secondly causes data contamination, and thirdly, the additional unknown items set might lead to unnecessary and unknown problems.

https://github.com/apache/eventmesh/blob/9ce5dd6fd94d60d47358721a18eff0f50cd932ad/eventmesh-storage-plugin/eventmesh-storage-kafka/src/main/java/org/apache/eventmesh/storage/kafka/producer/ProducerImpl.java#L51-L56

As for why RocketMQ and Pulsar modules don't need `clear()`, because it is consistent with their original logic.

Reply:
Then how about moving the `properties.put` statements of `org.apache.eventmesh.storage.kafka.producer.ProducerImpl` into the `init()` method of `org.apache.eventmesh.storage.kafka.producer.KafkaProducerImpl`?

Reply:
Actually, when we read the code of `ProducerImpl` and `KafkaProducerImpl` , it is quite evident that the latter is merely a delegate class, delegating all the specific details to former for actual processing. Of course, you can use the code from the corresponding classes in RocketMQ module to refute me. However, I personally favor the positioning of these two classes in Kafka module, as their responsibilities are more clearly defined.

Finally Community Over Code. I'm willing to wait for more members in community to express their views on this point. If the approach used in RocketMQ module is more accepted, I will revise.

Reply:
>it is quite evident that the latter is merely a delegate class, delegating all the specific details to former for actual processing

I agree with your view. Since the abstraction has already been made, how about standardizing the usage of the abstract implementation?

Reply:
My view on what you call 'standardizing' is similar to https://github.com/apache/eventmesh/pull/3867#discussion_r1576133454 above.